src/structarray.jl

```diff
@@ -338,12 +338,10 @@ to
 map(c -> c[I...], Tuple(cols))
 ```
 """
-@inline get_ith(cols::NamedTuple, I...) = get_ith(Tuple(cols), I...)
-@inline function get_ith(cols::Tuple, I...)
-    @inbounds r = first(cols)[I...]
-    return (r, get_ith(Base.tail(cols), I...)...)
-end
-@inline get_ith(::Tuple{}, I...) = ()
+@inline @generated get_ith(cols::Union{
+        NTuple{N, Any},
+        NamedTuple{<:Any,<:NTuple{N, Any}}
```

Comment:
I imagine you could also just dispatch on `cols::T where {T<:Tup}` and use `N = fieldcount(T)` to have a simpler dispatch. I'm always afraid that complex dispatch mechanisms might slow down the compiler, but I confess I only have a very vague understanding of that, so maybe it does not matter.